### DISCUSSION TITLE
If a page has two columns, how can I make all the footnotes on that page appear at the bottom of the right column?

### DISCUSSION THREAD

**Discussion by wcshds** [typst/typst#4810] [Q&A] [answered]
I want to place all the footnotes at the bottom of the right column to make the paper look neater, but it seems difficult to implement.

<img src="https://github.com/user-attachments/assets/f3d082bd-0fa8-445d-b3fa-415b76a24ca9" width="500">

**Comment by laurmaedje** [comment, marked as answer]
I don't think it's possible.

**Comment by alerque** [comment]
Use SILE ;-) Typst is awesome in a lot of ways, but it's concept of footnotes doesn't cover arbitrary frames and stealing arbitrary space form other places. SILE is a bit more finicky to use but it lets you do arbitrary things like this with frames.

**Reply by laurmaedje** [reply]
I've been thinking a lot about how to make layout more flexible lately and I'd be totally up for some friendly exchange/chat about the different approaches taken in SILE, Typst, and TeX.

**Reply by alerque** [reply]
The process of handling insertions (SILE's term for content being sent from one place but processed in another that causes the target frame to grow and steal space from something else) is both really cool and allows really useful layout stuff, and also a royal pain in the butt. I've pulled out quite a bit of my greying beard over it. Some of that has to do with problems with SILE's early architecture which relied on functions with side effects. We've been getting a lot of that worked out, but the `pushBack()` function that handles content that overflowed when space was stolen after it was used for something else is still a source of buggy behavior. Some of it is the problem space is just inherently messy and giving content flexibility just by definition opens up Pandora's box no matter how you slice it.

**Reply by laurmaedje** [reply]
Oh, that sounds very interesting. What's the best place to read up on that? The SILE Book? Or something else.

The thing with the side effects sounds a bit like your doing "trial typesetting" or relayout and that causes issues when you have side effects?
